src/components/MultiGestureCanvas/usePanGesture.js

```diff
@@ -0,0 +1,241 @@
+/* eslint-disable no-param-reassign */
+import {Gesture} from 'react-native-gesture-handler';
+import {runOnJS, useDerivedValue, useSharedValue, useWorkletCallback, withDecay, withSpring} from 'react-native-reanimated';
+import * as MultiGestureCanvasUtils from './utils';
+
+const PAN_DECAY_DECELARATION = 0.9915;
+
+const clamp = MultiGestureCanvasUtils.clamp;
+const SPRING_CONFIG = MultiGestureCanvasUtils.SPRING_CONFIG;
+
+const usePanGesture = ({
+    canvasSize,
+    contentSize,
+    panGestureRef,
+    pagerRef,
+    singleTap,
+    doubleTap,
+    zoomScale,
+    zoomRange,
+    totalScale,
+    totalOffsetX,
+    totalOffsetY,
+    panTranslateX,
+    panTranslateY,
+    isSwipingVertically,
+    isSwipingHorizontally,
+    onSwipeSuccess,
+    stopAnimation,
+}) => {
+    const zoomScaledContentWidth = useDerivedValue(() => contentSize.width * totalScale.value, [contentSize.width]);
+    const zoomScaledContentHeight = useDerivedValue(() => contentSize.height * totalScale.value, [contentSize.height]);
+
+    const previousTouch = useSharedValue(null);
+    // pan velocity to calculate the decay
```

Comment:
Please begin comments with a capital letter, and comments should also have an empty line before them (not sure why the linter isn't complaining).